File: src/main/java/com/abhinav/lms/notification/service/NotificationService.java
package com.abhinav.lms.notification.service;

import com.abhinav.lms.notification.dto.NotificationResponse;
import com.abhinav.lms.notification.entity.NotificationType;
import com.abhinav.lms.security.model.UserPrincipal;
import com.abhinav.lms.user.entity.User;
import org.springframework.data.domain.Page;

import java.util.UUID;

public interface NotificationService {

    void sendNotification(User recipient, String title, String message, NotificationType type, String referenceId, String referenceType);

    Page<NotificationResponse> getUserNotifications(UserPrincipal user, int page, int size);

    void markAsRead(UUID notificationId, UserPrincipal user);

    void markAllAsRead(UserPrincipal user);

    long getUnreadCount(UserPrincipal user);
}

File: lms-frontend/services/notificationService.js
import api from './api';

export const notificationService = {
  async getNotifications(page = 0, size = 10) {
    const response = await api.get(`/notifications`, {
      params: { page, size }
    });
    return response.data;
  },

  async getUnreadCount() {
    const response = await api.get(`/notifications/unread-count`);
    return response.data;
  },

  async markAsRead(id) {
    const response = await api.put(`/notifications/${id}/read`);
    return response.data;
  },

  async markAllAsRead() {
    const response = await api.put(`/notifications/read-all`);
    return response.data;
  }
};
export default notificationService;
